syncer/syncer.go

```diff
@@ -908,6 +926,9 @@ var (
 )
 
 func (s *Syncer) addJob(job *job) error {
+	s.waitTransactionLock.Lock()
```

Comment:
I think we'd better check `waitXIDJob` after L929-L930.
Suggest the following case:
1. run `runCancel()` and `s.waitTransactionLock.Unlock()` in L1470-L1471.
2. `syncer.Run` quits and `s.done` is closed.
3. run `syncer.stopSync` and `s.closeJobChans`
4. `syncer.addJob` sends a dml job to the closed job channel.

Reply:
SGTM. 

How about delete L1397-L1404? Because the job in jobQueue have been executed before `closeJobChan()`.